core/error/error_macros.h

```diff
@@ -34,6 +34,10 @@
 
 #include <atomic> // IWYU pragma: keep // Used in macro. We'd normally use `safe_refcount.h`, but that would cause circular includes.
 
+#ifdef _MSC_VER
+#include <intrin.h> // `__fastfail()`.
```

Comment:
Perhaps we can pack it into a CPP file and include intrin there?
It's a bit scary to have it included all over the code base.

Reply:
Note that it's already included a good number of places thanks to `hashfuncs.h` also including it, so in practice it shouldn't be disruptive. Still, it'd be easy enough to just add a new function that just calls to these "builtins" within the cpp file instead, so I'll test that out rq.

Reply:
If it's already everywhere else it might be worth a separate PR to cpp-ify it, doesn't have to be resolved now :)

Reply:
Yeah, looks like it'd be a bit invasive otherwise, as this would otherwise change the behavior for all non-MSVC compilers & necessitate a pretty hefty recompilation. Gonna just let the MSVC people (me) suffer this time